File: crates/rspack_swc_visitors/src/react.rs
use swc_core::ecma::transforms::react::Runtime;

#[derive(Debug, Clone, Default)]
pub struct ReactOptions {
  pub runtime: Option<Runtime>,
  pub import_source: Option<String>,
  pub pragma: Option<String>,
  pub pragma_frag: Option<String>,
  pub throw_if_namespace: Option<bool>,
  pub development: Option<bool>,
  pub use_builtins: Option<bool>,
  pub use_spread: Option<bool>,
  pub refresh: Option<bool>,
}

File: packages/rspack/src/builtin-loader/swc/react.ts
type RawReactOptions = {
	runtime?: "automatic" | "classic";
	importSource?: string;
	pragma?: string;
	pragmaFrag?: string;
	throwIfNamespace?: boolean;
	development?: boolean;
	useBuiltins?: boolean;
	useSpread?: boolean;
	refresh?: boolean;
};

function resolveReact(react: ReactOptions): RawReactOptions {
	return react ?? {};
}

type ReactOptions = RawReactOptions | undefined;

export { resolveReact };
export type { ReactOptions };
